### ISSUE TITLE
Removing and creating a new category in a code list

### ISSUE THREAD

**Issue by HayleyMills** [CLOSER-Cohorts/archivist#77]
When you have removed a category from a code list and then enter a new one it moves the box up to the top, when you save it has moved it back to the bottom. Also, not all the up and down buttons work when doing this.

**Comment by HayleyMills** [comment]
This is still an ongoing problem and is causing issues with re-ordering code lists. Please see nshd_89_scq cs_sectionE_i_1-24_Y as an example where we cannot put the categories in the correct order.

**Comment by HayleyMills** [comment]
Tested on staging using https://closer-archivist-staging.herokuapp.com/instruments/1033/build/code_lists/36024 it will now move the categories up and down correctly which is great, but it does not save the new order. 

Also, when you remove a category this works, however when you add a new category this goes to the top of the code list instead of the bottom. See screenshots.
![image](https://user-images.githubusercontent.com/16664703/62282228-1a64c480-b447-11e9-9397-895ecaa19bd6.png)
![image](https://user-images.githubusercontent.com/16664703/62282238-1d5fb500-b447-11e9-8cc2-f65df3e3fce7.png)
![image](https://user-images.githubusercontent.com/16664703/62282250-22246900-b447-11e9-8048-52eeca2f3c79.png)

**Comment by HayleyMills** [comment]
@spuddybike was this tested and is it working now?

**Comment by spuddybike** [comment]
@HayleyMills sorry - overly zealous - reopened for testing

**Comment by HayleyMills** [comment]
Tested on staging using https://closer-archivist-staging.herokuapp.com/instruments/1033/build/code_lists/36024 and it will not save any of the changes made- removing a category or moving the order of the categories.

**Comment by simonreed** [comment]
same issue as commented here https://github.com/CLOSER-Cohorts/archivist/issues/298#issuecomment-518556180

tldr; it's not on staging yet.

**Comment by simonreed** [comment]
@HayleyMills can you please retest, this is now on staging.

**Comment by HayleyMills** [comment]
Tested on staging using alspac_03_aam and the re-ordered categories are saving. 

Remaining issue is that when you remove a category and then add a new one, it appears at the top, this then causes issues with using the Up and Down (you can't move the new category down, nor the second one up) as it seems to be treating the category like it is actually at the bottom of the list?

**Comment by simonreed** [comment]
@HayleyMills This is now fixed on staging

![Screen Capture on 2019-08-06 at 15-46-53](https://user-images.githubusercontent.com/13276/62550350-0283ab00-b862-11e9-86ef-017949c369bc.gif)

**Comment by HayleyMills** [comment]
Great. Tested on staging and this looks like its working okay as your video shows.

**Comment by HayleyMills** [comment]
Tested in staging by JS and this is still problematic for larger code lists e.g bcs_80_pi code list cs_qC1. When you add a new code list and enter the code value it moves it up, if you save it it saves in the correct position at the bottom. 

It works on small code lists.

**Comment by HayleyMills** [comment]
@simonreed is this ready to be tested now?

**Comment by simonreed** [comment]
This isn't on staging yet

**Comment by simonreed** [comment]
This is now on staging @HayleyMills

**Comment by HayleyMills** [comment]
Tested on staging using bcs_80_pi code list cs_qC1- when I starting adding the code value (11), one I type 1 it moves it up? See snip below. 

![image](https://user-images.githubusercontent.com/16664703/64704514-56f3f900-d4a6-11e9-9b56-f7b853fa1d91.png)

**Comment by simonreed** [comment]
@HayleyMills sorry this wasn't deployed to staging after all. I've just retested and this now works for  `https://closer-archivist-staging.herokuapp.com/instruments/992/build/code_lists/34724`

**Comment by HayleyMills** [comment]
Tested on staging with the same example above and it is now working as expected.
